tests/lib/state_test.py

```diff
@@ -697,3 +698,35 @@ def _route_sort_key(route):
     return (route.get(Route.TABLE_ID, Route.USE_DEFAULT_ROUTE_TABLE),
             route.get(Route.NEXT_HOP_INTERFACE, ''),
             route.get(Route.DESTINATION, ''))
+
+
+def test_merge_config_dns():
+    dns_config = _get_test_dns_config()
+    current_state = state.State({DNS.KEY: dns_config})
+    desire_state = state.State({})
+
+    desire_state.merge_dns_config(current_state)
+
+    assert current_state.config_dns == desire_state.config_dns
```

Comment:
Not sure if we need to test this, it looks to me like the following assert is the expected end result which includes this one.